sources/packages/api/src/services/application/application.service.ts

```diff
@@ -164,6 +166,88 @@ export class ApplicationService extends RecordDataModelService<Application> {
     );
   }
 
+  /**
+   * Updates Assessment status.
+   * @param applicationId application id to be updated.
+   * @param status status of the Assessment.
+   * An assessment need to happen, when the application_status is in assessment.
+   * @returns assessment status update result.
+   */
+  async updateAssessmentStatus(
+    applicationId: number,
+    status: AssessmentStatus,
+  ): Promise<UpdateResult> {
+    return this.repo.update(
+      { id: applicationId },
+      {
+        assessmentStatus: status,
+      },
+    );
+  }
+
+  /**
+   * Updates Confirmation of Enrollment(COE) status.
+   * @param applicationId application id to be updated.
+   * @param status status of the Confirmation of Enrollment.
+   * Confirmation of Enrollment need to happen, when the application_status is in enrollment.
+   * @returns COE status update result.
+   */
+  async updateCOEStatus(
+    applicationId: number,
+    status: COEStatus,
+  ): Promise<UpdateResult> {
+    return this.repo.update(
+      { id: applicationId },
+      {
+        coeStatus: status,
+      },
+    );
+  }
+
+  /**
+   * Updates overall Application status.
+   * @param applicationId application id to be updated.
+   * @param status status of the Application.
+   * @returns COE status update result.
+   */
+  async updateApplicationStatus(
+    applicationId: number,
+    status: ApplicationStatus,
+  ): Promise<UpdateResult> {
+    return this.repo.update(
+      { id: applicationId },
+      {
+        applicationStatus: status,
+      },
+    );
+  }
+
+  /**
+   * Updates overall Application status.
+   * @param applicationId application id to be updated.
+   * @param status status of the Application.
+   * @returns COE status update result.
+   */
+  async studentConfirmAssessment(applicationId: number): Promise<any> {
```

Comment:
I got the idea to reuse the same methods to updates the statuses, but IMO, executing them in parallel will not ensure that the update on DB happens successfully for all of them. One of them could fail and leave the record in an undesired state.
I would recommend do a simple update altogether or execute them in a transaction (I would prefer the first one 😉).

Reply:
Just to clarify, the Promise.all will wait for all promises to be returned, but it will not ensure at any moment that all had success or that all fail or that one fails and two worked.